src/components/WatchLater.jsx

```diff
@@ -0,0 +1,42 @@
+import { useSelector, useDispatch } from 'react-redux'
+import { Link } from 'react-router-dom'
+import watchLaterSlice from '../data/watchLaterSlice'
+import Movie from './Movie'
+import '../styles/starred.scss'
+
+const WatchLater = ({viewTrailer}) => {
```

Comment:
- To stick to the Single Responsibility Principle (SRP), we should split the movie listing and footer logic into distinct components.

- Start by creating a WatchLaterMovieList.jsx component.
- Then, create a WatchLaterFooter.jsx component.
- Ensure that the WatchLater component serves as a container that brings these components together.